creating-virtual-machines/virtualized-hardware-configuration.adoc

```diff
@@ -134,6 +134,43 @@ spec:
         claimName: myclaim
 ----
 
+Enabling cpu compatibility enforcement
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable the cpu compatibility enforcement, user may expand the
+`feature-gates` field in the kubevirt-config config map by adding the
+`CPUNodeDiscovery` to it.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubevirt-config
+  namespace: kubevirt
+  labels:
+    kubevirt.io: ""
+data:
+  feature-gates: "CPUNodeDiscovery"
+...
+----
+
+This feature-gate allows kubevirt to take VM cpu model and cpu features and create node selectors from them.
+With these node selectors, VM can be scheduled on the node which can support VM cpu model and features.
+
+To properly label the node, user can use (only for cpu models)
+https://github.com/ksimon1/cpu-node-labeller[cpu-node-labeller] in combination with
+https://github.com/ksimon1/cpu-model-nfd-plugin[cpu-model-nfd-plugin] or create node labels by himself. 
+Labels for features user has to create by himself.
+
+To install cpu-node-labeller to cluster, run `oc apply -f cpu-node-labeller.yaml` with 
+https://github.com/ksimon1/cpu-node-labeller/blob/master/cpu-node-labeller.yaml[this deploy manifest]
+It will deploy cpu-node-labeller with cpu-model-nfd-plugin. 
+
+cpu-model-nfd-plugin uses libvirt to get all supported cpu models on host and 
+cpu-node-labeller create labels from cpu models. Then Kubevirt can schedule VM 
+on node which has support for VM cpu model.
+
```

Comment:
we probably should mention if models/features are not case sensitive

Reply:
models are case sensitive afaik

Reply:
sorry, I ment case sensitive. (btw, we should check that this is what nfd/labeller does)

Reply:
labeller doesn't do anything with names which plugins reports. It just creates labels with whatever it gets (without any modifications).

Reply:
so need to check how the plugin reports

Reply:
cpu model plugin takes models from libvirt and passes them to labeller.